packages/libp2p/src/connection-manager/index.ts

```diff
@@ -66,6 +66,12 @@ export interface ConnectionManagerInit {
    */
   autoDialMaxQueueLength?: number
 
+  /**
+   * When we've failed to dial a peer, do not autodial them again within this
+   * number of ms. (default: 10 minutes)
+   */
+  autoDialPeerRetryThreshold?: number
```

Comment:
honestly, even 1 minute would significantly help from my testing experience.

Once we add in a badPeer/bad multiaddr, it will free up time to check others instead of constantly spamming the same addresses.